src/Tribe/Views/V2/iCalendar/Traits/Outlook_Methods.php

```diff
@@ -85,78 +136,87 @@ protected function generate_outlook_add_url_parameters( $calendar = 'live' ) {
 		 */
 		$include_event_description = (bool) apply_filters( 'tec_events_ical_outlook_include_event_description', true );
 
-		if ( $include_event_description ) {
-			$body = $event->post_content;
-
-			// Stripping tags
-			$body = strip_tags( $body, '<p>' );
-
-			// Truncate Event Description and add permalink if greater than 900 characters
-			if ( strlen( $body ) > 900 ) {
+		if ( ! $include_event_description ) {
+			return '';
+		}
 
-				$body = substr( $body, 0, 900 );
+		$body = $event->post_content;
 
-				$event_url = get_permalink( $event->ID );
+		// Stripping most tags.
+		$body = wp_kses_post( $body );
 
-				//Only add the permalink if it's shorter than 900 characters, so we don't exceed the browser's URL limits (~2000)
-				if ( strlen( $event_url ) < 900 ) {
-					$body .= ' ' . sprintf( esc_html__( '(View Full %1$s Description Here: %2$s)', 'the-events-calendar' ), tribe_get_event_label_singular(), $event_url );
-				}
+		/**
+		 * Allows filtering the content of the event description.
+		 * Note: This happens before space conversion and truncation happens!
+		 *
+		 * @since TBD
+		 *
+		 * @param string $body The event description.
+		 */
+		$body = apply_filters( 'tec_events_ical_outlook_event_description_content', $body );
+
+		// Truncate the Event description and add permalink if greater than 900 characters.
+		if ( strlen( $body ) > 900 ) {
```

Comment:
```suggestion
		if ( strlen( $body ) > $filtered_body_length ) {
```
Shouldn't we filter this so people can override? Seems like the type of thing we shouldn't hard code.

Reply:
This is an old value and my understanding is it's a limit of the API - perhaps we should revisit it (later)?